fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPartitionUtil.java

```diff
@@ -504,4 +511,45 @@ public static Type getPartitionColumnType(MTMVRelatedTableIf relatedTable, Strin
         }
         throw new AnalysisException("can not getPartitionColumnType by:" + col);
     }
+
+    public static List<String> getPartitionsByRange(MTMV mtmv, MTMVRefreshPartitionRange range) {
+        List<String> res = Lists.newArrayList();
+        Collection<Partition> partitions = mtmv.getPartitions();
+        for (Partition partition : partitions) {
+            PartitionItem item = mtmv.getPartitionInfo().getItem(partition.getId());
+            if (ifPartitionItemInRange(item, range)) {
+                res.add(partition.getName());
+            }
+        }
+        return res;
+    }
+
+    private static boolean ifPartitionItemInRange(PartitionItem item, MTMVRefreshPartitionRange range) {
+        if (item instanceof ListPartitionItem) {
+            ListPartitionItem listPartitionItem = (ListPartitionItem) item;
+            List<PartitionKey> keys = listPartitionItem.getItems();
+            for (PartitionKey key : keys) {
+                if (ifPartitionKeyInRange(key, range, false)) {
+                    return true;
+                }
+            }
+        } else if (item instanceof RangePartitionItem) {
+            RangePartitionItem rangePartitionItem = (RangePartitionItem) item;
+            Range<PartitionKey> items = rangePartitionItem.getItems();
+            return ifPartitionKeyInRange(items.lowerEndpoint(), range, false) || ifPartitionKeyInRange(
+                    items.lowerEndpoint(), range, true);
+        }
+        return false;
+    }
+
+    private static boolean ifPartitionKeyInRange(PartitionKey partitionKey, MTMVRefreshPartitionRange range,
+            boolean isUpper) {
+        Preconditions.checkState(partitionKey.getKeys().size() == 1,
+                "only support one partition column");
+        String stringValue = partitionKey.getKeys().get(0).getStringValue();
+        DateV2Literal dateV2Literal = new DateV2Literal(stringValue);
```

Comment:
i think u'd better throw a clarify error message. not use DateV2Literal init error mesasge